File: C++/day1p1.cc
#include <iostream>
#include <fstream>

int main() {
    std::ifstream f{"input1.txt"}; // Input file
    std::string line = ""; // Line in file
    int ans = 0; // Answer

    // Read lines from the file
    while (std::getline(f, line)) {
        int first = 0, last = 0; // First and last digits
        bool fdigit = true; // Is it the first digit
        int len = line.length(); // Get length

        // Iterate through line
        for (int i = 0; i < len; ++i) {
            if (isdigit(line[i])) { // If the current character is a digit
                if (fdigit) first = line[i] - '0'; // Update first if it is the first digit
                last = line[i] - '0'; // Always update last
                fdigit = false; // No longer the first digit
            }
        }
        ans += ((10 * first) + last); // Update the final answer
    }

    std::cout << ans << std::endl; // Print out the final answer
}

File: Python/day1p1.py
ans = 0 # Answer

# Open input file ans iterate through lines
with open("input1.txt", "r") as f:
    for line in f.readlines():
        first = 0; last = 0 # First and last digits
        fdigit = True # Is it the first digit
        for c in line: # Iterate through the characters in the line
            if c.isnumeric(): # If the character is a digit
                if fdigit: first = int(c) # Update first if it is the first digit
                last = int(c) # Always update last
                fdigit = False # No longer the first digit
        ans += ((10 * first) + last) # Update the final answer

print(ans) # Print out the final answer
